src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -36,12 +36,12 @@ public void Complete(IContainSagaData saga)
             var values = data.Values.Where(x => x.SagaEntity is TSagaData);
             foreach (var entity in values)
             {
-                var prop = entity.SagaEntity.GetType().GetProperty(propertyName);
+                var prop = typeof(TSagaData).GetProperty(propertyName);
                 if (prop == null)
                 {
                     continue;
                 }
-                if (!prop.GetValue(entity.SagaEntity, null).Equals(propertyValue))
+                if (!Equals(prop.GetValue(entity.SagaEntity, null), propertyValue))
```

Comment:
yeah it is always weird that u dont need to do `Object.Equals` but because u dont need the static class when u inherit Equals just works.